### DISCUSSION TITLE
Skip going to https://repo1.maven.org/maven2

### DISCUSSION THREAD

**Discussion by TomasMikula** [VirtusLab/scala-cli#2835] [Using directives and cmd configuration options]
Hi,

I am trying Scala CLI in a walled environment without direct access to Maven Central.

I have managed to run a simple script with a library dependency using `--repo https://my.company/repo` and also alternatively using the directive
```
//> using repositories https://my.company/repo
```

However, I can see that **on each run** (via `scala-cli --server=false hello.sc`), Scala-CLI first attempts to go `https://repo1.maven.org/maven2`.

**Is it possible to disable going to Maven Central altogether?**

Note that setting

```
scala-cli --power config repositories.default https://my.company/repo
```

did not help.

Also, why doesn't local cache kick in before trying Maven Central?

**Comment by adpi2** [comment]
In Coursier there is a [--no-default](https://get-coursier.io/docs/2.1.0-M3-1/other-repositories#docsNav) option to ignore the default repositories (Maven Central and the Ivy 2 local repo). It is probably a good idea to add it to Scala CLI too.

**Comment by jiramares** [comment]
Hi, I am using enviroment variable export COURSIER_REPOSITORIES and it looks that maven central is not used at all, eg:
COURSIER_REPOSITORIES="ivy2local|http://our-local-server/repo"

**Reply by TomasMikula** [reply]
Thanks for the suggestion. For now, I have given up on making Scala CLI work smoothly in a walled environment, it's too much trouble.

However, having a _library_ (in this case Coursier) reading environment variables is exactly the kind of thing that leads to configuration hell. I don't want to have to know about Coursier.
